### PR TITLE
Index datahub from sepcstore - refs #28

### PR DESCRIPTION
**Pull request by zelima**
* small module for indexing
* grab descriptor for dataset via quering S3
* run ES and S3 server on Travis
* tests

### CODE REVIEW

File: flowmanager/datasets.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+tpe = ThreadPoolExecutor(max_workers=1)
+
+ELASTICSEARCH_HOST = os.environ.get('EVENTS_ELASTICSEARCH_HOST', 'localhost:9200')
+DATASETS_INDEX_NAME = os.environ.get('DATASETS_INDEX_NAME', 'datahub')
+DATASETS_DOCTYPE = 'dataset'
+
+SCHEMA = {
+    'fields': [
+        {'name': 'id', 'type': 'string'},
```
Comment (akariv) [inline flowmanager/datasets.py]: we should start using different analyzers for different fields:
https://www.elastic.co/guide/en/elasticsearch/reference/5.3/analysis-analyzers.html

e.g. for id, name etc. I would use keyword analyzer

File: flowmanager/datasets.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    ]}}
+             ]}
+         },
+    ],
+    'primaryKey': ['id']
+}
+
+
+def normalize(obj):
+    if isinstance(obj, (dict, LazyJsonLine)):
```
Comment (akariv) [inline flowmanager/datasets.py]: not needed.
which types of these are actually possible? the data written comes from a json object.

File: flowmanager/datasets.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        "name": name,
+        "title": title,
+        "description": description,
+        "datahub": datahub,
+        "datapackage": datapackage
+    }
+
+    primary_key = SCHEMA['primaryKey']
+    storage = Storage(es)
+    storage.create(
```
Comment (akariv) [inline flowmanager/datasets.py]: custom mapper (for setting custom analyzers) comes here

File: flowmanager/datasets.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        DATASETS_INDEX_NAME, [(DATASETS_DOCTYPE, SCHEMA)], always_recreate=False)
+
+    list(storage.write(DATASETS_INDEX_NAME, DATASETS_DOCTYPE, [normalize(body)],
```
Comment (akariv) [inline flowmanager/datasets.py]: no point in using as_generator=True and list together...

File: flowmanager/datasets.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                             primary_key, as_generator=True))
+
+class EventSender():
```
Comment (akariv) [inline flowmanager/datasets.py]: why `EventSender`?

File: flowmanager/datasets.py [hunk 6]
```diff
@@ -16,22 +16,22 @@
 
 SCHEMA = {
     'fields': [
-        {'name': 'id', 'type': 'string'},
-        {'name': 'name', 'type': 'string'},
-        {'name': 'title', 'type': 'string'},
-        {'name': 'description', 'type': 'string'},
+        {'name': 'id', 'type': 'string', 'analyzer': 'keyword'},
```
Comment (akariv) [inline flowmanager/datasets.py]: this won't set the analyzer. You need to override the mapper class here https://github.com/datahq/specstore/pull/29/commits/d3bc547e5c76b07e1573890fd3b27b27e4b326f5#r160045579

**Comment by zelima** [comment]
@akariv after latest changes the mapping looks like following

```
// curl http://localhost:9200/datahub/_mapping

"datahub_20180108_173c40b0": {
  "mappings": {
    "dataset": {
      "properties": {
        "datahub": {
          "dynamic": "false",
          "properties": {
            "findability": {
              "type": "text",
              "analyzer": "keyword"
            },
            "flowid": {
              "type": "text",
              "analyzer": "keyword"
            },
            "owner": {
              "type": "text",
              "analyzer": "keyword"
            },
            "ownerid": {
              "type": "text",
              "analyzer": "keyword"
            },
            "stats": {
              "dynamic": "false",
              "properties": {
                "bytes": {
                  "type": "long",
                  "index": false,
                  "ignore_malformed": true
                },
                "rowcount": {
                  "type": "long",
                  "index": false,
                  "ignore_malformed": true
                }
              }
            }
          }
        },
        "datapackage": {
          "dynamic": "false",
          "properties": {
            "readme": {
              "type": "text",
              "analyzer": "standard"
            }
          }
        },
        "description": {
          "type": "text",
          "analyzer": "standard"
        },
        "id": {
          "type": "text",
          "analyzer": "keyword"
        },
        "name": {
          "type": "text",
          "analyzer": "keyword"
        },
        "title": {
          "type": "text",
          "analyzer": "simple"
        }
      }
    }
  }
}

```

**Pull request merged**